### ISSUE TITLE
ignoreByDefault = true with explicit mentioned mapping only intializes that mapping, but does not map subfields.

### ISSUE THREAD

**Issue by Zegveld** [mapstruct/mapstruct#3577]
### Expected behavior

given the situation in the reproduction area I would expect the following result, since the mapping `@Mapping(source="sub", target="target")` is explicitly present.

```
@Generated(
    value = "org.mapstruct.ap.MappingProcessor",
    date = "2024-04-25T13:10:57+0200",
    comments = "version: 1.6.0-SNAPSHOT, compiler: javac, environment: Java 17.0.4.1 (Azul Systems, Inc.)"
)
class MyMapperImpl implements MyMapper {

    @Override
    public Target map(Source source) {
        if ( source == null ) {
            return null;
        }

        Target target = new Target();

        target.setTarget( subSourceToSubTarget( source.getSub() ) );

        return target;
    }

    protected SubTarget subSourceToSubTarget(SubSource subSource) {
        if ( subSource == null ) {
            return null;
        }

        SubTarget subTarget = new SubTarget();

        subTarget.setChildren( sourceChildListToTargetChildList( subSource.getChildren() ) );

        return subTarget;
    }

    protected TargetChild sourceChildToTargetChild(SourceChild sourceChild) {
        if ( sourceChild == null ) {
            return null;
        }

        TargetChild targetChild = new TargetChild();

        targetChild.setField( sourceChild.getField() );

        return targetChild;
    }

    protected List<TargetChild> sourceChildListToTargetChildList(List<SourceChild> list) {
        if ( list == null ) {
            return null;
        }

        List<TargetChild> list1 = new ArrayList<TargetChild>( list.size() );
        for ( SourceChild sourceChild : list ) {
            list1.add( sourceChildToTargetChild( sourceChild ) );
        }

        return list1;
    }
}

```

### Actual behavior

```
@Generated(
    value = "org.mapstruct.ap.MappingProcessor",
    date = "2024-04-25T13:09:22+0200",
    comments = "version: 1.6.0-SNAPSHOT, compiler: javac, environment: Java 17.0.4.1 (Azul Systems, Inc.)"
)
class MyMapperImpl implements MyMapper {

    @Override
    public Target map(Source source) {
        if ( source == null ) {
            return null;
        }

        Target target = new Target();

        target.setTarget( subSourceToSubTarget( source.getSub() ) );

        return target;
    }

    protected SubTarget subSourceToSubTarget(SubSource subSource) {
        if ( subSource == null ) {
            return null;
        }

        SubTarget subTarget = new SubTarget();

        return subTarget;
    }
}
```

### Steps to reproduce the problem

```
// leaving out getters/setters to keep it compact.
class Source {
  SubSource sub;
  // lot of other fields
}

class SubSource {
  List<SourceChild> children;
}

class SourceChild {
  String field;
}

class Target {
  SubTarget target;
}

class SubTarget {
  List<TargetChild> children;
}

class TargetChild {
  String field;
}

@Mapper
interface MyMapper {
  @BeanMapping(ignoreByDefault=true)
  @Mapping(source="sub", target="target")
  Target map(Source source);
}
```

### MapStruct Version

Mapstruct 1.6.0.Beta1

**Comment by Zegveld** [comment]
We also need to keep in mind that when using nested mappings the usage of `ignoreByDefault = true` should last until we arrive at the explicitly defined mapping.

**Comment by filiphr** [comment]
I'm not sure that we can treat this as a bug @Zegveld. What if you have
```java
@Mapper
interface MyMapper {
  @BeanMapping(ignoreByDefault=true)
  @Mapping(source="sub.field", target="target.field")
  Target map(Source source);
}
```

Should everything in be mapped or only that particular field. When using `ignoreByDefault` I think that it is OK to expect people to define deep nested mappings if they want to customize things more.

**Comment by Zegveld** [comment]
Currently I am forced to create the following:
```java
@Mapper
interface MyMapper {
  @BeanMapping(ignoreByDefault=true)
  @Mapping(source="sub", target="target")
  Target map(Source source);
  
  // this is the mapping `source = "sub"` and `target = "target"` 
  // redefined to get rid of the `ignoreByDefault=true` even though this is the explicitly defined mapping above.
  SubTarget subSourceToSubTarget(SubSource subSource);
}
``` 
because otherwise the the mapping here is not created correctly.

> I'm not sure that we can treat this as a bug @Zegveld. What if you have
> ```
> @Mapper
> interface MyMapper {
>   @BeanMapping(ignoreByDefault=true)
>   @Mapping(source="sub.field", target="target.field")
>   Target map(Source source);
> }
> ```
>Should everything in be mapped or only that particular field. When using ignoreByDefault I think that it is OK to expect people to define deep nested mappings if they want to customize things more.
>

In case of you example I would expect only that `field` **and all the subfields of that class** to be mapped.
The result would be that only `field` gets mapped.

**Comment by Zegveld** [comment]
Just as an example I've created some test classes with several nested mappings.

Just showing the mapper change and the result change in the generated mapper. For more information about the code see the example on github.

The following mapper is without `ignoreByDefault=true`
```
@Mapper
public interface DeepNestedMapper {
    @Mapping( target = "target.nestedSecondTargetChild.autoMapChild", source = "source.sourceInnerChild.nestedSecondSourceChild" )
    @Mapping( target = "target.nestedTargetChild", source = "source.sourceInnerChild.nestedSourceChild" )
    @Mapping( target = "targetCollection", source = "collectionContainer.source" )
    @BeanMapping( unmappedTargetPolicy = ReportingPolicy.ERROR)
    TargetContainer map(SourceContainer source);
}
```
this results in the following mapper being generated:
```
@Generated(
    value = "org.mapstruct.ap.MappingProcessor",
    date = "2024-04-27T13:32:45+0200",
    comments = "version: , compiler: javac, environment: Java 17.0.3 (Eclipse Adoptium)"
)
public class DeepNestedMapperImpl implements DeepNestedMapper {

    @Override
    public TargetContainer map(SourceContainer source) {
        if ( source == null ) {
            return null;
        }

        TargetContainer targetContainer = new TargetContainer();

        targetContainer.setTarget( sourceToTarget( source.getSource() ) );
        List<SourceCollectionItem> source1 = sourceCollectionContainerSource( source );
        targetContainer.setTargetCollection( sourceCollectionItemListToTargetCollectionItemList( source1 ) );

        return targetContainer;
    }

    protected TargetSubChild sourceSecondSubChildToTargetSubChild(SourceSecondSubChild sourceSecondSubChild) {
        if ( sourceSecondSubChild == null ) {
            return null;
        }

        TargetSubChild targetSubChild = new TargetSubChild();

        targetSubChild.setAutoMapField( sourceSecondSubChild.getAutoMapField() );

        return targetSubChild;
    }

    protected TargetChild sourceInnerChildToTargetChild(SourceInnerChild sourceInnerChild) {
        if ( sourceInnerChild == null ) {
            return null;
        }

        TargetChild targetChild = new TargetChild();

        targetChild.setAutoMapChild( sourceSecondSubChildToTargetSubChild( sourceInnerChild.getNestedSecondSourceChild() ) );

        return targetChild;
    }

    private List<SourceChild> sourceSourceInnerChildNestedSourceChild(Source source) {
        SourceInnerChild sourceInnerChild = source.getSourceInnerChild();
        if ( sourceInnerChild == null ) {
            return null;
        }
        return sourceInnerChild.getNestedSourceChild();
    }

    protected TargetSubChild sourceSubChildToTargetSubChild(SourceSubChild sourceSubChild) {
        if ( sourceSubChild == null ) {
            return null;
        }

        TargetSubChild targetSubChild = new TargetSubChild();

        targetSubChild.setAutoMapField( sourceSubChild.getAutoMapField() );

        return targetSubChild;
    }

    protected TargetChild sourceChildToTargetChild(SourceChild sourceChild) {
        if ( sourceChild == null ) {
            return null;
        }

        TargetChild targetChild = new TargetChild();

        targetChild.setAutoMapChild( sourceSubChildToTargetSubChild( sourceChild.getAutoMapChild() ) );

        return targetChild;
    }

    protected List<TargetChild> sourceChildListToTargetChildList(List<SourceChild> list) {
        if ( list == null ) {
            return null;
        }

        List<TargetChild> list1 = new ArrayList<TargetChild>( list.size() );
        for ( SourceChild sourceChild : list ) {
            list1.add( sourceChildToTargetChild( sourceChild ) );
        }

        return list1;
    }

    protected Target sourceToTarget(Source source) {
        if ( source == null ) {
            return null;
        }

        Target target = new Target();

        target.setNestedSecondTargetChild( sourceInnerChildToTargetChild( source.getSourceInnerChild() ) );
        List<SourceChild> nestedSourceChild = sourceSourceInnerChildNestedSourceChild( source );
        target.setNestedTargetChild( sourceChildListToTargetChildList( nestedSourceChild ) );

        return target;
    }

    private List<SourceCollectionItem> sourceCollectionContainerSource(SourceContainer sourceContainer) {
        SourceCollectionContainer collectionContainer = sourceContainer.getCollectionContainer();
        if ( collectionContainer == null ) {
            return null;
        }
        return collectionContainer.getSource();
    }

    protected TargetCollectionItem sourceCollectionItemToTargetCollectionItem(SourceCollectionItem sourceCollectionItem) {
        if ( sourceCollectionItem == null ) {
            return null;
        }

        TargetCollectionItem targetCollectionItem = new TargetCollectionItem();

        targetCollectionItem.setItem( sourceCollectionItem.getItem() );

        return targetCollectionItem;
    }

    protected List<TargetCollectionItem> sourceCollectionItemListToTargetCollectionItemList(List<SourceCollectionItem> list) {
        if ( list == null ) {
            return null;
        }

        List<TargetCollectionItem> list1 = new ArrayList<TargetCollectionItem>( list.size() );
        for ( SourceCollectionItem sourceCollectionItem : list ) {
            list1.add( sourceCollectionItemToTargetCollectionItem( sourceCollectionItem ) );
        }

        return list1;
    }
}
```

Now the same Mapper, but with `ignoreByDefault=true`. Since all fields are defined, this should not change anything.
```
@Mapper
public interface DeepNestedMapper {
    @Mapping( target = "target.nestedSecondTargetChild.autoMapChild", source = "source.sourceInnerChild.nestedSecondSourceChild" )
    @Mapping( target = "target.nestedTargetChild", source = "source.sourceInnerChild.nestedSourceChild" )
    @Mapping( target = "targetCollection", source = "collectionContainer.source" )
    @BeanMapping( ignoreByDefault = true )
    TargetContainer map(SourceContainer source);
}
```
this results in the following mapper being generated:
```
package org.mapstruct.ap.test.nestedbeans.deepstructure;

import java.util.ArrayList;
import java.util.List;
import javax.annotation.processing.Generated;

@Generated(
    value = "org.mapstruct.ap.MappingProcessor",
    date = "2024-04-27T13:42:41+0200",
    comments = "version: , compiler: javac, environment: Java 17.0.3 (Eclipse Adoptium)"
)
public class DeepNestedMapperImpl implements DeepNestedMapper {

    @Override
    public TargetContainer map(SourceContainer source) {
        if ( source == null ) {
            return null;
        }

        TargetContainer targetContainer = new TargetContainer();

        targetContainer.setTarget( sourceToTarget( source.getSource() ) );
        List<SourceCollectionItem> source1 = sourceCollectionContainerSource( source );
        targetContainer.setTargetCollection( sourceCollectionItemListToTargetCollectionItemList( source1 ) );

        return targetContainer;
    }

    protected TargetSubChild sourceSecondSubChildToTargetSubChild(SourceSecondSubChild sourceSecondSubChild) {
        if ( sourceSecondSubChild == null ) {
            return null;
        }

        TargetSubChild targetSubChild = new TargetSubChild();

        return targetSubChild;
    }

    protected TargetChild sourceInnerChildToTargetChild(SourceInnerChild sourceInnerChild) {
        if ( sourceInnerChild == null ) {
            return null;
        }

        TargetChild targetChild = new TargetChild();

        targetChild.setAutoMapChild( sourceSecondSubChildToTargetSubChild( sourceInnerChild.getNestedSecondSourceChild() ) );

        return targetChild;
    }

    private List<SourceChild> sourceSourceInnerChildNestedSourceChild(Source source) {
        SourceInnerChild sourceInnerChild = source.getSourceInnerChild();
        if ( sourceInnerChild == null ) {
            return null;
        }
        return sourceInnerChild.getNestedSourceChild();
    }

    protected TargetChild sourceChildToTargetChild(SourceChild sourceChild) {
        if ( sourceChild == null ) {
            return null;
        }

        TargetChild targetChild = new TargetChild();

        return targetChild;
    }

    protected List<TargetChild> sourceChildListToTargetChildList(List<SourceChild> list) {
        if ( list == null ) {
            return null;
        }

        List<TargetChild> list1 = new ArrayList<TargetChild>( list.size() );
        for ( SourceChild sourceChild : list ) {
            list1.add( sourceChildToTargetChild( sourceChild ) );
        }

        return list1;
    }

    protected Target sourceToTarget(Source source) {
        if ( source == null ) {
            return null;
        }

        Target target = new Target();

        target.setNestedSecondTargetChild( sourceInnerChildToTargetChild( source.getSourceInnerChild() ) );
        List<SourceChild> nestedSourceChild = sourceSourceInnerChildNestedSourceChild( source );
        target.setNestedTargetChild( sourceChildListToTargetChildList( nestedSourceChild ) );

        return target;
    }

    private List<SourceCollectionItem> sourceCollectionContainerSource(SourceContainer sourceContainer) {
        SourceCollectionContainer collectionContainer = sourceContainer.getCollectionContainer();
        if ( collectionContainer == null ) {
            return null;
        }
        return collectionContainer.getSource();
    }

    protected TargetCollectionItem sourceCollectionItemToTargetCollectionItem(SourceCollectionItem sourceCollectionItem) {
        if ( sourceCollectionItem == null ) {
            return null;
        }

        TargetCollectionItem targetCollectionItem = new TargetCollectionItem();

        return targetCollectionItem;
    }

    protected List<TargetCollectionItem> sourceCollectionItemListToTargetCollectionItemList(List<SourceCollectionItem> list) {
        if ( list == null ) {
            return null;
        }

        List<TargetCollectionItem> list1 = new ArrayList<TargetCollectionItem>( list.size() );
        for ( SourceCollectionItem sourceCollectionItem : list ) {
            list1.add( sourceCollectionItemToTargetCollectionItem( sourceCollectionItem ) );
        }

        return list1;
    }
}
```

comparing the 2 results:

```patch
@@ -6,7 +6,7 @@ import javax.annotation.processing.Generated;

 @Generated(
     value = "org.mapstruct.ap.MappingProcessor",
-    date = "2024-04-27T13:32:45+0200",
+    date = "2024-04-27T13:42:41+0200",
     comments = "version: , compiler: javac, environment: Java 17.0.3 (Eclipse Adoptium)"
 )
 public class DeepNestedMapperImpl implements DeepNestedMapper {
@@ -33,6 +33,8 @@ public class DeepNestedMapperImpl implements DeepNestedMapper {

         TargetSubChild targetSubChild = new TargetSubChild();

-        targetSubChild.setAutoMapField( sourceSecondSubChild.getAutoMapField() );
-
         return targetSubChild;
     }

@@ -56,6 +58,18 @@ public class DeepNestedMapperImpl implements DeepNestedMapper {
         return sourceInnerChild.getNestedSourceChild();
     }

-    protected TargetSubChild sourceSubChildToTargetSubChild(SourceSubChild sourceSubChild) {
-        if ( sourceSubChild == null ) {
-            return null;
-        }
-
-        TargetSubChild targetSubChild = new TargetSubChild();
-
-        targetSubChild.setAutoMapField( sourceSubChild.getAutoMapField() );
-
-        return targetSubChild;
-    }
-
     protected TargetChild sourceChildToTargetChild(SourceChild sourceChild) {
         if ( sourceChild == null ) {
             return null;
@@ -63,6 +77,8 @@ public class DeepNestedMapperImpl implements DeepNestedMapper {

         TargetChild targetChild = new TargetChild();

-        targetChild.setAutoMapChild( sourceSubChildToTargetSubChild( sourceChild.getAutoMapChild() ) );
-
         return targetChild;
     }

@@ -108,6 +124,8 @@ public class DeepNestedMapperImpl implements DeepNestedMapper {

         TargetCollectionItem targetCollectionItem = new TargetCollectionItem();

-        targetCollectionItem.setItem( sourceCollectionItem.getItem() );
-
         return targetCollectionItem;
     }
```

**Comment by filiphr** [comment]
I'm not sure that I agree with you @Zegveld.

Why should

```java
  @BeanMapping(ignoreByDefault=true)
  Target map(Source source);
```

Not map any of the defined fields, but

```java
  @BeanMapping(ignoreByDefault=true)
  @Mapping(source="sub", target="target")
  Target map(Source source);
```

Should perform deep nested mapping

and

```java
  @BeanMapping(ignoreByDefault=true)
  @Mapping(source="sub.field", target="target.field")
  Target map(Source source);
```

should only map the `target.field` and not the rest of the fields in `target`.

`BeanMapping#ignoreByDefault` is inherited to the deep nested mappings. If you want something else then you'll have to define the specific mapping method and configure it

**Comment by Zegveld** [comment]
Because those are **explicit mappings**.

See the sections below as to why I come to that conclusion:
----
The information that I find about `ignoreByDefault` says the following:
> By means of the @BeanMapping(ignoreByDefault = true) the default behavior will be explicit mapping, meaning that all mappings have to be specified by means of the @Mapping and no warnings will be issued on missing target properties. This allows to ignore all fields, except the ones that are explicitly defined through @Mapping.

If I read the above description of `ignoreByDefault` it shouldn't be inherited into explicit mappings, because they have been explicitly defined....
However I agree that it should be inherited into implicit mappings, because those have not been explicitly defined....

Examples:
`@Mapping(source="fieldA", target="fieldB")` 
I have explicitly stated that I want to map all contents of `fieldA` to `fieldB`.

`@Mapping(source="fieldA.subField", target="fieldB.subField")`
I have explicitly stated that I want to map all contents of property `subField` of `fieldA` to property `subField` of `fieldB`.
I have implicitly stated that `fieldA` and `fieldB` are related, but not that it should map any other properties here, because my only explicit mapping is the `subfield` one.

----
Looking at it from another perspective:

The following is an **explicit mapping**, correct?
`@Mapping(source="something", target="otherthing")`
The default behavior of this **explicit mapping** is that it maps all the properties of these beans if possible, correct?

The description of `ignoreByDefault` mentions `the default behavior will be explicit mapping`. So when this is enabled the above should still happen, because that is an **explicit mapping**. Or did the definition of an explicit mapping change as soon as I enabled `ignoreByDefault`?

**Comment by filiphr** [comment]
>The following is an explicit mapping, correct?
>`@Mapping(source="something", target="otherthing")`
>The default behavior of this explicit mapping is that it maps all the properties of these beans if possible, correct?

No the behavior is that the `otherthing` should be mapped into `something` it does not mean that all properties should be mapped. You might have a dedicated mapper for mapping those types and then it will still be mapped becacuse you have *explicitly* defined it.

The main question here is what should the MapStruct generated nested mapping inherit from `@BeanMapping`

If we ignore `ignoreByDefault` for a second and we have the following

```java
class Target {

    NestedTarget nested;
}

class NestedTarget {

    String name;
}

class Source {

    NestedSource nested;
}

class NestedSource {

    String firstName;
}
```

And we have the following mapper:

```java
@Mapper
public interface MyMapper {

    @BeanMapping(unmappedTargetPolicy = ReportingPolicy.ERROR)
    Target map(Source source);

}
```

then the `unmappedTargetPolicy` is inherited to the nested mappings as well and there will be a compile error. The `ignoreByDefault` is doing exactly the same, it propagates the property to the nested mappings. If we change that then it means we are changing the inheritance of the `@BeanMapping`.

Basically the following mappers are identical:

```java
@Mapper
public interface MyMapper {

    @Mapping(target = "nested")
    @BeanMapping(ignoreByDefault = true)
    Target map(Source source);

}
```

```java
@Mapper
public interface MyMapper {

    @Mapping(target = "nested")
    @BeanMapping(ignoreByDefault = true)
    Target map(Source source);

    @BeanMapping(ignoreByDefault = true)
    NestedTarget map(NestedSource source);

}
```

**Comment by Zegveld** [comment]
@filiphr, what is an explicit mapping, and what should the behavior of an explicit mapping be?
I think my definition of this is different, and this is where the trouble lies with understanding each other.

Also for 1.5.5.Final the behavior was not inherited, so this is a change which should end up in the change log for 1.6.0 if we keep the behavior as it is now.
At work our mappers broke because of this change when upgrading from 1.5.5.Final to 1.6.0.Beta1.
Gonna see if I can figure out which change was the cause for this behavior change.

**Comment by filiphr** [comment]
>@filiphr, what is an explicit mapping, and what should the behavior of an explicit mapping be?
I think my definition of this is different, and this is where the trouble lies with understanding each other.

An explicit mapping is a mapping defined with `@Mapping`. The fact that the defined mapping is an object does not mean that we should automatically drill down in the properties. My main point is about the fact whether we should be propagating the `@BeanMapping` to the forged methods or not.

>Also for 1.5.5.Final the behavior was not inherited, so this is a change which should end up in the change log for 1.6.0 if we keep the behavior as it is now.

The fix we did https://github.com/mapstruct/mapstruct/issues/3158 for caused this. However, I believe that what we are doing in 1.6.0.Beta1 is correct and what was there in 1.5.5.Final is a bug. We should not be applying implicit mappings when applying explicit mapping. My main point is that `@BeanMapping` should be applied to all of the downstream forged mappings.

Therefore, I believe that 

```java
@Mapper
public interface MyMapper {

    @Mapping(target = "nested")
    @BeanMapping(ignoreByDefault = true)
    Target map(Source source);

}
```

and 

```java
@Mapper
public interface MyMapper {

    @Mapping(target = "nested")
    @BeanMapping(ignoreByDefault = true)
    Target map(Source source);

    @BeanMapping(ignoreByDefault = true)
    NestedTarget map(NestedSource source);

}
```

should be treated as identical.

Otherwise, #3135 should be reverted. As it is the exact same point. The `BeanMapping#mappingControl` was not being propagated to the forged methods.

----

The inheritance of the `@BeanMapping` is what we should be discussing, not particular properties of it. Why should `ignoreByDefault` not be propagated, but `mappingControl` should?

If you look at 

https://github.com/mapstruct/mapstruct/blob/8e66445fe9f13b37ea66191548067a80bcc6d048/processor/src/main/java/org/mapstruct/ap/internal/model/source/BeanMappingOptions.java#L58-L67

You'll see that we are propagating all the parameters except the `resultType` and `ignoreUnmappedSourceProperties` which are only applicable for the single mapping method.

**Comment by Zegveld** [comment]
> The fact that the defined mapping is an object does not mean that we should automatically drill down in the properties.

In this case I would suggest that we update the documentation to contain a better explanation of the `ignoreByDefault` behavior. The following could be interpreted as only affecting that mapping method:
> By means of the @BeanMapping(ignoreByDefault = true) the default behavior will be explicit mapping, meaning that all mappings have to be specified by means of the `@Mapping` and no warnings will be issued on missing target properties. This allows to ignore all fields, except the ones that are explicitly defined through `@Mapping`.

**Comment by filiphr** [comment]
I'm fine with improving it. I would rather do something on the top level of `@BeanMapping` and then for the properties that are not inherited in the forged methods mention it.

Otherwise, we need to add something to all the other properties, as 

>Allows detailed control over the mapping process

>How unmapped properties of the target type of a mapping should be reported.

>How unmapped properties of the source type of a mapping should be reported.

>Determines when to include a null check on the source property value of a bean mapping.

can also be interpreted as only affecting that mapping method. I guess you understand what I'm trying to say.

**Comment by Zegveld** [comment]
I've pushed an attempt at improving the documentation at both the `BeanMapping` class and the asciidoc.

**Comment by filiphr** [comment]
Thanks @Zegveld. It look good, I've added one small comment only. I'll also change the name of the issue if that's OK with you

**Comment by Zegveld** [comment]
Updated it according to your comment, and feel free to rename the issue.

**Comment by filiphr** [comment]
Thanks @Zegveld. I've merged the PR

**Comment by Zegveld** [comment]
@filiphr, the change in behavior that was noticed here might be important enough to add to the `1.6.0.Beta1` under the `Breaking Changes` section.

It was in `1.5.5.Final` unintended behavior, but those who use it now will see it as an unexpected change if it is not listed there. Perhaps even with an example of how to correctly define the wanted behavior of continuing with implicit mappings.

For example we could add the following to the `Breaking Changes` section:
BeanMapping properties are designed by default to be inherited, this includes `ignoreByDefault`. In previous versions the inheritence of this property didn't behave as expected.

To get the same result as in previous versions you need to change the following:
```
@Mapper
interface MyMapper {
  @BeanMapping( ignoreByDefault = true )
  @Mapping( source = "sub", target = "target" )
  Target map( Source source );
}
```
into this
```
@Mapper
interface MyMapper {
  @BeanMapping( ignoreByDefault = true )
  @Mapping( source = "sub", target = "target" )
  Target map( Source source );
  
  // this is the mapping `source = "sub"` and `target = "target"` 
  // redefined to get rid of the `ignoreByDefault=true`, because this property is inherited
  SubTarget subSourceToSubTarget( SubSource subSource );
}
```

**Comment by filiphr** [comment]
I've updated the release notes
